docs/src/pages/solid.mdx

```diff
@@ -258,44 +244,89 @@ const Home: VoidComponent = () => {
       />
     </main>
   );
-};
-
-export default Home;
+}
 ```
 
-#### UploadDropzone
+#### Uploader
 
-This is a SolidJS dropzone component that you can use to upload files, you can
-drag files into it.
+A combination of the `UploadButton` and `UploadDropzone` components.
 
 ```tsx copy
-// You need to import our styles for the button to look right. Best to import in the root /layout.tsx but this is fine
-import "@uploadthing/solid/styles.css";
-
-import { type VoidComponent } from "solid-js";
-
-import { UploadDropzone } from "@uploadthing/solid";
+// Or import from your typed, generated components
+import { Uploader } from "@uploadthing/solid";
 
 import type { OurFileRouter } from "~/server/uploadthing";
 
-const Home: VoidComponent = () => {
+export default function Home() {
   return (
-    <main class="flex min-h-screen flex-col items-center justify-center  gap-16 p-24">
-      <UploadDropzone<OurFileRouter>
-        endpoint="imageUploader"
+    <main>
+      <Uploader<OurFileRouter>
+        endpoint="profileImage"
         // needed when server side rendering
         url="http://localhost:3000"
         onClientUploadComplete={() => {
           alert("Upload Completed");
         }}
-        onUploadError={(error: Error) => {
-          // Do something with the error.
-          alert(`ERROR! ${error.message}`);
-        }}
       />
     </main>
   );
-};
+}
+```
+
+## Advanced usage: `useUploadThing`
+
+For advanced use cases, the premade components might not be flexible enough to
+suit your needs. In that case, you can use the `useUploadThing` hook to build
+your own components with a custom upload flow:
+
+<Steps>
+
+### Create the hook
+
+First, generate a typed hook using the `generateSolidHelpers` function from
+`@uploadthing/solid`:
 
-export default Home;
+```tsx copy filename="src/utils/uploadthing.ts"
+import { generateSolidHelpers } from "@uploadthing/solid";
+
+import type { OurFileRouter } from "~/server/uploadthing";
+
+export const { useUploadThing } = generateSolidHelpers<OurFileRouter>();
 ```
+
+### Use the hook
+
+Then, use the hook in your component. By using the hook, you have full control
+of when and how to upload files.
+
+```tsx copy filename="src/routes/index.tsx"
+import { useUploadThing } from "~/utils/uploadthing";
+
+export default function Home() {
+  const { startUpload } = useUploadThing("profileImage", {
+    onClientUploadComplete: () => {
+      alert("Upload Completed");
+    },
+  });
+
+  return (
+    <main>
+      <input
+        type="file"
+        onChange={async (e) => {
+          const file = e.target.files?.[0];
+          if (!file) return;
+
+          // Do something with files
+          const compressed = await compress(file);
```

Comment:
since this is copyable code block, we should probably either have a dummy `compress` function, or do something different here.